internal/evaluation_target/application_snapshot_image/application_snapshot_image.go

```diff
@@ -298,48 +298,80 @@ func (a *ApplicationSnapshotImage) Attestations() []attestation.Attestation {
 	return a.attestations
 }
 
-func (a *ApplicationSnapshotImage) AttestationsOutput() []output.Attestation {
-	var attestations []output.Attestation
-	for _, att := range a.attestations {
-		attestations = append(attestations, att.Output())
+func (a *ApplicationSnapshotImage) Signatures() []signature.EntitySignature {
+	return a.signatures
+}
+
+type attestationData struct {
+	json.RawMessage
+	Extra attestationExtraData `json:"extra"`
+}
+
+type attestationExtraData struct {
+	Signatures []signature.EntitySignature `json:"signatures,omitempty"`
+}
+
+func (a attestationData) MarshalJSON() ([]byte, error) {
+	buffy := bytes.Buffer{}
+	raw, err := a.RawMessage.MarshalJSON()
+	if err != nil {
+		return nil, err
 	}
-	return attestations
+
+	_, err = buffy.Write(raw[0 : len(raw)-1])
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = buffy.WriteString(`,"extra":`)
+	if err != nil {
+		return nil, err
+	}
+
+	extra, err := json.Marshal(a.Extra)
+	if err != nil {
+		return nil, err
+	}
+
+	buffy.Write(extra)
+	buffy.WriteByte('}')
```

Comment:
I would expect `{` earlier in this function. Is `a.RawMessage.MarshalJSON()` returning a partial JSON?

Reply:
[`a.RawMessage.MarshalJSON()`](https://github.com/enterprise-contract/ec-cli/pull/829/files/ba2b2404a6d64e3d7856e379319c8d3185be9bd7#diff-c89fbe58faac75c0a251f1b8af58b7789adc10532e0814ad59ef0adc47d91445R316) is returning JSON from which the last `}` [is stripped](https://github.com/enterprise-contract/ec-cli/pull/829/files/ba2b2404a6d64e3d7856e379319c8d3185be9bd7#diff-c89fbe58faac75c0a251f1b8af58b7789adc10532e0814ad59ef0adc47d91445R321).

Reply:
Aha! Missed that.